graphql/ui_plugin_build_baron.go

```diff
@@ -8,17 +8,113 @@ import (
 	"time"
 
 	"github.com/evergreen-ci/evergreen"
+	"github.com/evergreen-ci/evergreen/model/event"
+	"github.com/evergreen-ci/evergreen/model/notification"
 	"github.com/evergreen-ci/evergreen/model/task"
 	"github.com/evergreen-ci/evergreen/thirdparty"
+	"github.com/evergreen-ci/evergreen/trigger"
+	"github.com/evergreen-ci/evergreen/units"
+	"github.com/evergreen-ci/utility"
 	"github.com/mitchellh/mapstructure"
 	"github.com/mongodb/grip"
 	"github.com/pkg/errors"
 )
 
 const (
-	jiraSource = "JIRA"
+	jiraSource    = "JIRA"
+	jiraIssueType = "Build Failure"
 )
 
+// bbFileTicket creates a JIRA ticket for a task with the given test failures.
+func BbFileTicket(taskId string) (bool, error) {
+	taskNotFound := false
```

Comment:
The callers of this don't really use the return value. Would it be sufficient to have this function just return an error, and have the callers pass that error along?

Reply:
This is needed so the resolver knows wether or not to return ResourceNotFound or InternalServerError. I can remove the boolean and send an InternalServerError if the task is not found, but I don't think that would be an improvement.